lisa/transformers/upgrade_packages.py

```diff
@@ -0,0 +1,170 @@
+from dataclasses import dataclass, field
+from typing import Any, Dict, List, Optional, Type
+
+from dataclasses_json import dataclass_json
+
+from lisa import schema
+from lisa.node import Node, quick_connect
+from lisa.operating_system import Debian
+from lisa.tools import Uname
+from lisa.transformer import Transformer
+from lisa.util import field_metadata, subclasses
+from lisa.util.logger import Logger, get_logger
+
+
+@dataclass_json()
+@dataclass
+class BaseInstallerSchema(schema.TypedSchema, schema.ExtendableSchemaMixin):
```

Comment:
If nothing to abstract at this layer, it doesn't need to declare a baseinstaller layer. The kernel installer derives to package installer and source installer. For this repo installer, it doesn't need to have the base installer part. And the same "BaseInstaller" name is confusing.

Reply:
I changed some class names and schema names.